src/bun.js/api/html_rewriter.zig

```diff
@@ -1117,85 +1036,64 @@ pub const TextChunk = struct {
-            },
-
-            .publicId = .{
-                .get = getterWrap(DocType, "publicId"),
-            },
-        },
-    );
+    pub usingnamespace JSC.Codegen.JSDocType;
 
     /// The doctype name.
-    pub fn name(this: *DocType, global: *JSGlobalObject) JSValue {
+    pub fn name(
+        this: *DocType,
+        globalObject: *JSGlobalObject,
+    ) callconv(.C) JSValue {
         if (this.doctype == null)
-            return JSC.JSValue.jsUndefined();
+            return JSValue.jsUndefined();
         const str = this.doctype.?.getName().slice();
         if (str.len == 0)
             return JSValue.jsNull();
-        return ZigString.init(str).toValueGC(global);
+        return ZigString.init(str).toValueGC(globalObject);
     }
 
-    pub fn systemId(this: *DocType, global: *JSGlobalObject) JSValue {
+    pub fn systemId(
+        this: *DocType,
+        globalObject: *JSGlobalObject,
+    ) callconv(.C) JSValue {
         if (this.doctype == null)
-            return JSC.JSValue.jsUndefined();
+            return JSValue.jsUndefined();
 
         const str = this.doctype.?.getSystemId().slice();
         if (str.len == 0)
             return JSValue.jsNull();
-        return ZigString.init(str).toValueGC(global);
+        return ZigString.init(str).toValueGC(globalObject);
     }
 
-    pub fn publicId(this: *DocType, global: *JSGlobalObject) JSValue {
+    pub fn publicId(
+        this: *DocType,
+        globalObject: *JSGlobalObject,
+    ) callconv(.C) JSValue {
         if (this.doctype == null)
-            return JSC.JSValue.jsUndefined();
+            return JSValue.jsUndefined();
 
         const str = this.doctype.?.getPublicId().slice();
         if (str.len == 0)
             return JSValue.jsNull();
-        return ZigString.init(str).toValueGC(global);
+        return ZigString.init(str).toValueGC(globalObject);
     }
 };
 
 pub const DocEnd = struct {
     doc_end: ?*LOLHTML.DocEnd,
 
-    pub fn finalize(this: *DocEnd) void {
+    pub fn finalize(this: *DocEnd) callconv(.C) void {
         this.doc_end = null;
         bun.default_allocator.destroy(this);
     }
 
-    pub const Class = NewClass(
-        DocEnd,
-        .{ .name = "DocEnd" },
-        .{
-            .finalize = finalize,
-            .append = .{
-                .rfn = wrap(DocEnd, "append"),
-            },
-        },
-        .{},
-    );
+    pub usingnamespace JSC.Codegen.JSDocEnd;
 
-    fn contentHandler(this: *DocEnd, comptime Callback: (fn (*LOLHTML.DocEnd, []const u8, bool) LOLHTML.Error!void), thisObject: js.JSObjectRef, globalObject: *JSGlobalObject, content: ZigString, contentOptions: ?ContentOptions) JSValue {
+    fn contentHandler(this: *DocEnd, comptime Callback: (fn (*LOLHTML.DocEnd, []const u8, bool) LOLHTML.Error!void), callFrame: *JSC.CallFrame, globalObject: *JSGlobalObject, content: ZigString, contentOptions: ?ContentOptions) JSValue {
         if (this.doc_end == null)
```

Comment:
one thing to be aware of is that `callFrame` is a pointer to a stack frame and things can get moved around as the stack frame changes a little 

So we should prefer to reference all variables from the call frame at the beginning of the callframe, and not pass it around to non-inline functions

In this case, it's super cheap to get the `this` value. It's a pointer lookup without other side effects